### ISSUE TITLE
Add "In Progress" Status

### ISSUE THREAD

**Issue by adamzolotarev** [exceptionless/Exceptionless.UI#51]
When someone starts working on an exception, I would like to mark it as "In Progress" in Exceptionless and be able to search by this status.
Not sure if "In Progress" would be clear enough. Maybe, "Being addressed". 

Or maybe add a way to add any custom status

**Comment by niemyjski** [comment]
We have support for pushing stacks to third party boards like trello, and you can set it up so when the card is moved to done it auto marks the stack as fixed. This allows you to use exceptionless with a more stand up functionality.  It would be nice to have some kind of status and assign a user. I personally think there is a fine line to where you get into an issue tracker and we want to do a few things really well. What are your thoughts on this?

**Comment by adamzolotarev** [comment]
I think I'm moving more toward pushing a status to a third party system when a new stack is created. Then having some job that would push updates from a third party (tfs server in my case) back to exceptionless. I'm assuming you are talking about webhooks for pushing events to third parties. Do you have a code example of how a webhook could be implemented?

**Comment by niemyjski** [comment]
Yes, we use zapier.com for most third party integrations. But if you want to call your own api, you'd just need to go to the integrations tab of your project and add a new hook for when a new event is created. What this will do is post to an api end point with a json blob of data.

Here is an example of what data will be posted: https://github.com/exceptionless/Exceptionless/tree/master/Source/Tests/Plugins/WebHookData you can take a look at our plugins to see exactly whats posted. If you have any questions, please don't hesitate to ask.

**Comment by niemyjski** [comment]
Here is the data we post if it's v1 or a v2 web hook (They are versioned): https://github.com/exceptionless/Exceptionless/tree/master/Source/Core/Plugins/WebHook/Default

Do you think we should add something to our docs for this?

**Comment by adamzolotarev** [comment]
I think it would be helpful to someone like me who just starts with Exceptionless. I'll try to add something as I go along. Thank you!

**Comment by adamzolotarev** [comment]
I think I still need a way in the UI to say that something is promoted to external system. Maybe there is a better way than adding another status. 
I couldn't find anywhere if there is an easy way to filter stacks which have/or don't have a reference link - this could be a way to say that someone is working on it.

**Comment by niemyjski** [comment]
I'll bring this up in tomorrows discussion and see what we can do. Off the top of my head you should be able to search for stacks that have reference links set by running the search query `_exists_:links` (I haven't personally tried this)

**Comment by adamzolotarev** [comment]
I tried it with both _exists_ and _missing_ ; doesn't seem to work. Although, it looks like there is an index "links". I'm trying to understand how the search terms from the UI are delivered to Elastic search. Maybe, that could be something worth implementing or fixing (if it is already supposed to work)

**Comment by niemyjski** [comment]
I was just able to reproduce this behavior. It's due to we are filtering the ui by events and not stacks of which this index exists on. We run the search terms through a query parser and then just pass it onto elastic search. We've had a request for status / notes and the ability to assign ownership. I've asked my team mate to look into this as well. I think it would be nice to have this kind of functionality as it would add value. For now as a work around, I think the best option is to use a web hook using zapper to control your existing story boards (it just stinks that you don't know immediately from the stack list which is assigned and which isn't). This is something we could improve on.

**Comment by adamzolotarev** [comment]
Discovered something interesting - you can search for _exists_:links , but the results only show up in New tab. The same applies when searching for regressed:true.

**Comment by niemyjski** [comment]
How are you searching in the new tab? Are you calling our api?

**Comment by adamzolotarev** [comment]
Sorry, shouldn't have called it a tab. Searching from Exceptionless UI. Enter "_exists_:links" or "regressed:true" in Search. Nothing is displayed in Dashboard, Most Recent, or Most Frequent pages. But when you go to "New", the result is displayed.
